### DISCUSSION TITLE
input `pattern` attribute

### DISCUSSION THREAD

**Discussion by AnthonyLenglet** [Devographics/surveys#172] [State of HTML Features]
### Code sample

```html
<label for="username">Username: (3-16 characters)</label>
<input name="username" type="text" value="Sasha" pattern="\w{3,16}" required>

<label for="pin">PIN: (4 digits)</label>
<input name="pin" type="password" pattern="\d{4,4}" required>
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern

### Specification URL:

https://html.spec.whatwg.org/multipage/input.html#attr-input-pattern

### Any other links?

https://caniuse.com/input-pattern

### How stable is this feature?

Stable (Widely supported for several years)

### Type of feature

HTML attribute

### What focus area does this feature fall under?

Forms

### Details

This is an attribute that seems like it would be really useful, but I don't recall ever seeing it outside of blog posts, it could be interesting to see if this is just due to a lack of knowledge about the attribute or something else that might be missing from it

**Comment by LeaVerou** [comment]
I wondered about this too. Right now this is covered by https://github.com/Devographics/surveys/discussions/120 , if we ask about `pattern` separately, we'd probably need to ask about many others separately, and it's a lot of questions for things that are quite stable in terms of browser support...

**Reply by AnthonyLenglet** [reply]
that would make sense actually, and seeing the reply there, I do agree with domenic as well, getting more details would probably be helpful

maybe there could be a question like "out of those form validation features, which ones do you use (/ use the most)", and a second one asking for the ones people would want to see get improved and in what capacity ?
